src/sentry/migrations/0100_collect_relay_analytics.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.29 on 2020-09-11 07:15
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+import django.utils.timezone
+import sentry.db.models.fields.bounded
+
+
+class Migration(migrations.Migration):
+    # This flag is used to mark that a migration shouldn't be automatically run in
+    # production. We set this to True for operations that we think are risky and want
+    # someone from ops to run manually and monitor.
+    # General advice is that if in doubt, mark your migration as `is_dangerous`.
+    # Some things you should always mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that
+    #   they can be monitored. Since data migrations will now hold a transaction open
+    #   this is even more important.
+    # - Adding columns to highly active tables, even ones that are NULL.
+    is_dangerous = False
+
+    # This flag is used to decide whether to run this migration in a transaction or not.
+    # By default we prefer to run in a transaction, but for migrations where you want
+    # to `CREATE INDEX CONCURRENTLY` this needs to be set to False. Typically you'll
+    # want to create an index concurrently when adding one to an existing table.
+    atomic = True
+
+
+    dependencies = [
+        ('sentry', '0099_fix_project_platforms'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='RelayUsage',
+            fields=[
+                ('id', sentry.db.models.fields.bounded.BoundedBigAutoField(primary_key=True, serialize=False)),
+                ('relay_id', models.CharField(max_length=64)),
+                ('version', models.CharField(default=b'0.0.1', max_length=32)),
+                ('first_seen', models.DateTimeField(default=django.utils.timezone.now)),
+                ('last_seen', models.DateTimeField(default=django.utils.timezone.now)),
+            ],
+            options={
+                'db_table': 'sentry_relayusage',
+            },
+        ),
+        migrations.RemoveField(
+            model_name='relay',
+            name='first_seen',
+        ),
+        migrations.RemoveField(
+            model_name='relay',
+            name='last_seen',
```

Comment:
Is this model currently receiving read or write traffic in production? If so dropping the column like this will cause errors. Migrations are run before code is deployed. After this migration has run Django will continue to try and use these columns in any read/write operations it does which will fail.

To solve the above problem we [ship drop migrations in multiple pieces](https://develop.sentry.dev/database-migrations/#columns).

Reply:
first_seen is never used and last_seen is only _set_, never read. Attribute writes on models always succeed regardless of whether the field is defined.

Jan can assess the impact here if this breaks but I think this might be safe. Note that we will have to rely on this for st/onprem anyway unless *both* just shut everything down while migrating.

Reply:
Good point, this would cause Relay registration to temporarily stop functioning but it shouldn't be a problem in practice.

Relay does try to re register on failures that are not client errors (`HTTP 4xx`) so any Relay that was going to register immediately after migrations were run but before sentry was redeployed would receive `HTTP 5xx` and would be in the `not ready` state until Sentry deployment finishes.

I have made the requested changes to follow the guide, and will roll the changes to the `Relay` model in stages.